src/HtmlGenerator/Pass1-Generation/SolutionGenerator.cs

```diff
@@ -323,6 +326,7 @@ public bool Generate(HashSet<string> processedAssemblyList = null, Folder<Projec
 
             var projectsToProcess = allProjects
                 .Where(p => processedAssemblyList == null || processedAssemblyList.Add(p.AssemblyName))
+                .Where(p => !IsTestProject(p) || !ExcludeTests)

```

Comment:
Can you put the `!ExcludeTests` first, and then the IsTestProject check second? This way we don't have to run the expensive checks in the regular case for every project.

Reply:
done!